www/updateUes.php

```diff
@@ -6,20 +6,122 @@
-$body = json_decode($entityBody, true);
+$token = json_decode($entityBody, true);
 
-$attributesFromIdP = $body['attributes'];
-$attrMap = $body['attrMap'];
-$attrsToConversion = $body['attrsToConversion'];
-$perunUserId = $body['perunUserId'];
+$attributesFromIdP = null;
+$attrMap = null;
+$attrsToConversion = null;
+$perunUserId = null;
+$id = null;
 
 const UES_ATTR_NMS = 'urn:perun:ues:attribute-def:def';
+const TABLE_NAME = 'scriptChallenges';
+const CHALLENGE = 'challenge';
+const CONFIG_FILE_NAME = 'keys.php';
+
+try {
+    $config = Configuration::getConfig(CONFIG_FILE_NAME);
+    $keyPub = $config->getString('updateUes');
+
+    $algorithmManager = new AlgorithmManager([new RS512()]);
+    $jwsVerifier = new JWSVerifier($algorithmManager);
+    $jwk = JWKFactory::createFromKeyFile($keyPub);
+
+    $serializerManager = new JWSSerializerManager([new CompactSerializer()]);
+    $jws = $serializerManager->unserialize($token);
+
+    $headerCheckerManager = new HeaderCheckerManager([new AlgorithmChecker(['RS512'])], [new JWSTokenSupport()]);
+    $headerCheckerManager->check($jws, 0);
+
+    $isVerified = $jwsVerifier->verifyWithKey($jws, $jwk, 0);
+
+    if (!$isVerified) {
+        die('The token signature is invalid!');
+    }
+
+    $claimCheckerManager = new ClaimCheckerManager(
+        [
+            new Checker\IssuedAtChecker(),
+            new Checker\NotBeforeChecker(),
+            new Checker\ExpirationTimeChecker(),
+        ]
+    );
+
+    $claims = json_decode($jws->getPayload(), true);
+    $claimCheckerManager->check($claims);
+
+    $challenge = $claims['challenge'];
+
+    $attributesFromIdP = $claims['data']['attributes'];
+    $attrMap = $claims['data']['attrMap'];
+    $attrsToConversion = $claims['data']['attrsToConversion'];
+    $perunUserId = $claims['data']['perunUserId'];
+    $id = $claims['id'];
+
+    $databaseConnector = new DatabaseConnector();
+    $conn = $databaseConnector->getConnection();
+
+    assert($conn !== null);
+
+    $stmt = $conn->prepare('SELECT challenge FROM ' . TABLE_NAME . ' WHERE id=?');
+
+    if (!$stmt) {
+        die('Error during preparing statement');
+    }
+
+    $stmt->bind_param('s', $id);
+    $ex = $stmt->execute();
+
+    if ($ex === false) {
+        die('Error while getting the challenge from the database.');
+    }
+
+    $challengeDb = $stmt->get_result()->fetch_assoc()[CHALLENGE];
+
+    $stmt->close();
+
+    if (!hash_equals($challengeDb, $challenge)) {
+        die('Hashes are not equal.');
+    }
+
+    $stmt = $conn->prepare('DELETE FROM ' . TABLE_NAME . ' WHERE id=?');
+
+    if ($stmt) {
+        $stmt->bind_param('s', $id);
+        $ex = $stmt->execute();
+
+        if ($ex === false) {
+            Logger::error('Error while deleting the challenge from the database.');
+        }
+
+        $stmt->close();
+    } else {
+        Logger::error('Error during preparing statement');
+    }
+
+    $conn->close();
+} catch (Exception $ex) {
```

Comment:
[do not catch general Exception](https://stackoverflow.com/a/1743018) if not absolutely necessary